onnxruntime/core/providers/dml/DmlExecutionProvider/src/Operators/DmlOperatorQAttention.cpp

```diff
@@ -0,0 +1,703 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "precomp.h"
+
+/*
+Abbreviations: B is batch_size, S is sequence_length, W is hidden_size
+               N is number of attention heads, H is head size, and W=N*H
+               M is mask_index tensor
+
+M, A, B, C and P are Inputs
```

Comment:
>M [](http://example.com/codeflow?start=0&length=1)

Now that we are handling few extra cases, where M can be generated on the fly using Diagonal_Matrix1, it is not always true that `M` will be an input provided by the model.
Can we please remove `M` from this comment line and add a separate comment line explaining that `M` can be a input or can be generated during runtime?